### PR TITLE
Improvement in release perf test

### PR DESCRIPTION
**Pull request by heemin32**
### Description
Simply the process to run performance test for release

### Previous behavior
0. Update cluster setting with `"knn.algo_param.index_thread_qty" : 4`
1. Download hdf5 data
2. Modify test template with correct value of endpoint, port, path to json files, and path to hdf5 files manually.
3. Run knn-per-tool.py
4. Repeat step2 and step3 for 12 test cases

### After change
1. Download hdf5 data in k-NN/benchmakrs/perf-tool/dataset/
2. Run run_all_tests.sh --endpoint <your_endpoint>

### Issues Resolved
N/A
 
### Check List
- [X] New functionality includes testing.
  - [X] All tests pass
- [X] New functionality has been documented.
  - [X] New functionality has javadoc added
- [X] Commits are signed as per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/k-NN/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: benchmarks/perf-tool/release-configs/run_all_tests.sh [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+./release-configs/faiss-hnsw/test.yml
+./release-configs/faiss-hnswpq/test.yml
+./release-configs/faiss-ivf/filtering/relaxed-filter/relaxed-filter-test.yml
+./release-configs/faiss-ivf/filtering/restrictive-filter/restrictive-filter-test.yml
+./release-configs/faiss-ivf/test.yml
+./release-configs/faiss-ivfpq/test.yml
+./release-configs/lucene-hnsw/filtering/relaxed-filter/relaxed-filter-test.yml
+./release-configs/lucene-hnsw/filtering/restrictive-filter/restrictive-filter-test.yml
+./release-configs/lucene-hnsw/test.yml
+./release-configs/nmslib-hnsw/test.yml"
```
Comment (navneet1v) [inline benchmarks/perf-tool/release-configs/run_all_tests.sh:80]: can we add some sleep time between running of all the tests to make CPU come down and cluster to become stable.

Comment (heemin32) [reply benchmarks/perf-tool/release-configs/run_all_tests.sh:80]: > can we add some sleep time between running of all the tests to make CPU come down and cluster to become stable.

Do you think 1 min will be enough for that purpose?

Comment (navneet1v) [reply benchmarks/perf-tool/release-configs/run_all_tests.sh:80]: yeah 1 min is fine.

**Comment by navneet1v** [comment]
@heemin32 do you know why this `[k-NN Rolling-Upgrade BWC Tests (17, ubuntu-latest, 2.10.0-SNAPSHOT, 3.0.0-SNAPSHOT)](https://github.com/opensearch-project/k-NN/actions/runs/6489712314/job/17624422073?pr=1243#logs)` is coming even when we have removed to 2.10.0 from 2.10.0-SNAPSHOT? here: https://github.com/opensearch-project/k-NN/commit/34d697f7085789aabd4b1bf02d2aa33fe2fc6441

**Comment by heemin32** [comment]
> @heemin32 do you know why this `[k-NN Rolling-Upgrade BWC Tests (17, ubuntu-latest, 2.10.0-SNAPSHOT, 3.0.0-SNAPSHOT)](https://github.com/opensearch-project/k-NN/actions/runs/6489712314/job/17624422073?pr=1243#logs)` is coming even when we have removed to 2.10.0 from 2.10.0-SNAPSHOT? here: [34d697f](https://github.com/opensearch-project/k-NN/commit/34d697f7085789aabd4b1bf02d2aa33fe2fc6441)

https://github.com/opensearch-project/k-NN/pull/1245

**Comment by navneet1v** [comment]
> > @heemin32 do you know why this `[k-NN Rolling-Upgrade BWC Tests (17, ubuntu-latest, 2.10.0-SNAPSHOT, 3.0.0-SNAPSHOT)](https://github.com/opensearch-project/k-NN/actions/runs/6489712314/job/17624422073?pr=1243#logs)` is coming even when we have removed to 2.10.0 from 2.10.0-SNAPSHOT? here: [34d697f](https://github.com/opensearch-project/k-NN/commit/34d697f7085789aabd4b1bf02d2aa33fe2fc6441)
> 
> #1245

thanks for raising the PR

**Review by navneet1v — APPROVED** [review]

**Comment by martin-gaievski** [comment]
curious if with every next run time changes, as this all is on same cluster, I think some query stuff may be cached after multiple runs?
I think we may extend it in future by allowing different endpoints/clusters for different tests.

**Review by martin-gaievski — DISMISSED** [review]

**Comment by navneet1v** [comment]
> curious if with every next run time changes, as this all is on same cluster, I think some query stuff may be cached after multiple runs?
> I think we may extend it in future by allowing different endpoints/clusters for different tests.

@martin-gaievski queries won’t be cached because we delete the index at the start of every run. 
the only thing that may cause issue is RSS. But generally we run with 4xlarges so that shouldn’t not be a problem too

**Comment by heemin32** [comment]
> curious if with every next run time changes, as this all is on same cluster, I think some query stuff may be cached after multiple runs? I think we may extend it in future by allowing different endpoints/clusters for different tests.

This is what we have been doing for release benchmark test as long as I know. We are not using different cluster for each test run.

**Review by navneet1v — APPROVED** [review]

**Review by martin-gaievski — APPROVED** [review]

**Pull request merged**